src/excmds.ts

```diff
@@ -3072,7 +3072,38 @@ export async function bmark(url?: string, ...titlearr: string[]) {
     dupbmarks.map(bookmark => browser.bookmarks.remove(bookmark.id))
     if (dupbmarks.length != 0) return
     const path = title.substring(0, title.lastIndexOf("/") + 1)
-    // TODO: if title is blank, get it from the page.
+
+
+    //this whole block grabs the current tab's title
+    //based on an async call since document.title won't have any meaning here
+    //
+    var currTitle;
+    function getCurrTabTitle(tabs) {
+        //function callback for promise for getting title of current tab
+        //requires [tabs] permission which tridactyl should have grabbed anyway
+        if (tabs[0]) {
+            var currentTab = tabs[0];
+            currTitle = currentTab.title
+        }
+    }
+    var titleSourceTab = browser.tabs.query({active: true, currentWindow: true});
+    //TODO: define and implement sane behaviour for if the title is blank, but the URL
+    //      that is being bookmarked is not actually the current tab. just using 
+    //      await activeTab().title doesn't make sense for that, we'll need to change the tab query
+    //      immediately above to search for the url.  I guess then just go with arr[0] for the winner
+    //      The possibility also exists that someone tries to bmark a URL thats not a current tab anywhere
+    //      so sane behaviour needs to be defined for that too.  Load the page headless and extract URL?
+    //      Make the title the URL?
+    titleSourceTab.then(getCurrTabTitle);
+    await titleSourceTab;
```

Comment:
This is really hard to read - the `currTitle` side-effect is quite buried.

A clearer way of doing all of this would simply be:

```js
let currTitle = (await activeTab()).title
```

or, pretending you don't know about our own `activeTab`

```js
let currTitle = (
    await browser.tabs.query(
        {active: true, currentWindow: true}
    )
)[0].title
```

: )